SafetyFirst/NonEmpty.fs

```diff
@@ -21,22 +21,35 @@ type NonEmpty<'a, 'b when 'a :> 'b seq> = private NonEmpty of 'a with
 
 /// <summary>
 /// A seq constrained to be non-empty. 
-/// An alias for <c>NonEmpty<'a seq, 'a></c>
+/// An alias for <c>NonEmpty<'a seq, 'a></c>.
 /// </summary>
 type NonEmptySeq<'a> = NonEmpty<'a seq, 'a>
 
 /// <summary>
 /// An array constrained to be non-empty. 
-/// An alias for <c>NonEmpty<'a[], 'a></c>
+/// An alias for <c>NonEmpty<'a[], 'a></c>.
 /// </summary>
 type NonEmptyArray<'a> = NonEmpty<'a[], 'a>
 
 /// <summary>
 /// A list constrained to be non-empty. 
-/// An alias for <c>NonEmpty<'a list, 'a></c>
+/// An alias for <c>NonEmpty<'a list, 'a></c>.
 /// </summary>
 type NonEmptyList<'a> = NonEmpty<'a list, 'a>
 
+/// <summary>
+/// A set constrained to be non-empty. 
+/// An alias for <c>NonEmpty<Set<'a>, 'a></c>.
+/// </summary>
+type NonEmptySet<'a when 'a : comparison> = NonEmpty<Set<'a>, 'a>
+
+/// <summary>
+/// A Map constrained to be non-empty.
+/// An alias for <c>NonEmpty<Map<'key, 'value>, KeyValuePair<'key, 'value>></c>.
+/// </summary>
+type NonEmptyMap<'key, 'value when 'key : comparison> = 
+  NonEmpty<Map<'key, 'value>, KeyValuePair<'key, 'value>>
```

Comment:
maybe we shouldn't introduce an alias like this until we actually make a module for treating a NonEmptyMap like a regular map

Reply:
✔️